### PR TITLE
Integrate ResultComputation into the Voting contract.

### LINKED ISSUE #432: Integrate the ResultComputation library with the Voting contract

### PR DESCRIPTION
**Pull request by mrice32**
This PR introduces the mode computation into the Voting library such that true voting results can be computed. Since the token is still missing, all votes are equally weighted and all accounts effectively get a single vote.

Notes: 
- As a part of this PR, I restructured the voting contract a bit so that `getPrice()` and `hasPrice()` use the same internal function.
- In the process of testing, I found a preexisting bug in the `getPendingRequests()` function. It was previously undetected because no requests were being rolled over.
- I changed the name of `ResultComputationData` to `Data` since it's already in the `ResultComputation` namespace. This makes the name a bit shorter for other contracts that use it.

Fixes #432.

### CODE REVIEW

File: core/contracts/Voting.sol [hunk 1]
```diff
@@ -290,7 +331,7 @@ contract Voting is Testable, MultiRole, OracleInterface {
 
         // Allocate enough space for all of the price requests to be rolled over and just track the length
         // separately.
-        PriceRequest[] memory tmpRolloverArray = new PriceRequest[](numPriceRequests);
+        rolloverPriceRequests = new PriceRequest[](numPriceRequests);
```
Comment (ptare) [inline core/contracts/Voting.sol]: lol how did this ever work?

File: core/test/Voting.js [hunk 2]
```diff
@@ -415,4 +485,107 @@ contract("Voting", function(accounts) {
...
+    const hash3 = web3.utils.soliditySha3(winningPrice, salt3);
+    await voting.commitVote(identifier, time, hash3, { from: account3 });
+
+    // Reveal the votes.
+    await moveToNextPhase();
+    await voting.revealVote(identifier, time, losingPrice, salt1, { from: account1 });
+    await voting.revealVote(identifier, time, winningPrice, salt2, { from: account2 });
+    await voting.revealVote(identifier, time, winningPrice, salt3, { from: account3 });
+
+    // Should not have the price since the vote was equally split.
```
Comment (ptare) [inline core/test/Voting.js]: Incorrect comment?

File: core/contracts/Voting.sol (hunk 1 above)
Comment (mrice32) [reply core/contracts/Voting.sol]: It didn't :)

File: core/test/Voting.js (hunk 2 above)
Comment (mrice32) [reply core/test/Voting.js]: Done.

**Pull request merged**